src/library-extender/dam/custom-dam-provider.ts

```diff
@@ -0,0 +1,131 @@
+    }
+
+    loadMediaSelector(damWrapper: HTMLElement, mediaEntityData: EntityData, allowMultiSelect: boolean): void {
+        // first validate the settings coming from the server
+        if (!this.areSettingsPropertiesValid()) {
+            this.error(CUSTOM_MEDIA_CANNOT_BE_LOADED);
+            return;
+        }
+
+        this.loadDynamicScript(this.getPropertyValue("ScriptUrl")).subscribe(() => {
+            // if script is loaded successfully 'cloudinary' variable must be initialized
+            if (typeof cloudinary === "undefined") {
+                this.error(CUSTOM_MEDIA_CANNOT_BE_LOADED);
+                return;
+            }
+
+            // create configuration specific for Cloudinary
+            const config = {
+                cloud_name: this.getPropertyValue("CloudName"),
+                api_key: this.getPropertyValue("ApiKey"),
+                multiple: allowMultiSelect,
+                inline_container: `.${damWrapper.className.replace(/\s/g, ".")}`,
+                remove_header: true,
+                integration: {
+                    type: "custom_progress_sitefinity_connector_for_frontify",
+                    platform: "admin app extensions",
+                    version: "1.0",
+                    environment: null
+                }
+            };
+
+            const handlers = {
+                insertHandler: this.insertHandler.bind(this),
+                errorHandler: this.error.bind(this)
+            };
+
+            try {
+                // instantiate and open widget
+                cloudinary.openMediaLibrary(config, handlers);
+            } catch (error) {
+                this.error(error);
+            }
+        }, () => {
+            this.error(CUSTOM_MEDIA_CANNOT_BE_LOADED);
+        });
+    }
+
+    private insertHandler(data): void {
+        this.mediaSelected();
+        const damAssets: DamAsset[] = [];
+        if (data.assets) {
+            data.assets.forEach(asset => {
+                // convert selected DAM assets to DamAsset interface
+                const damAsset: DamAsset = this.getDamAsset(asset);
+                damAssets.push(damAsset);
+            });
+
+            // pass converted assets to base class
+            this.assetsSelected(damAssets);
+        } else {
+            this.error("FAILED_TO_INSERT_MEDIA_ASSET");
+        }
+    }
+
+    private areSettingsPropertiesValid(): boolean {
+        if (!this.settings) {
+            return false;
+        }
+
+        const cloudName = this.getPropertyValue("CloudName");
+        if (!cloudName) {
+            return false;
+        }
+
+        const scriptUrl = this.getPropertyValue("ScriptUrl");
+        if (!scriptUrl) {
+            return false;
+        }
+
+        const apiKey = this.getPropertyValue("ApiKey");
+        if (!apiKey) {
+            return false;
+        }
+
+        return true;
+    }
+
+    private getDamAsset(asset: any): DamAsset {
```

Comment:
The implementation is not up to date with the actual implementation for Cloudinary. If we want people to be able to use it as a reference, perhaps we need other changes as well.

Reply:
code is now updated to respect transformations and add sf_cl_ query params in the stored URL